test/src/kbasesearchengine/test/events/handler/WorkspaceEventHandlerTest.java

```diff
@@ -696,4 +703,528 @@ private void failLoadWSGetWSInfoException(final Exception toThrow, final Excepti
+
+        StatusEvent event = StatusEvent.getBuilder(
+                StorageObjectType.fromNullableVersion("WS", "foo", 1),
+                Instant.ofEpochMilli(20000),
+                StatusEventType.RENAME_ALL_VERSIONS)
+                .withNullableNewName("newName")
+                .withNullableAccessGroupID(1)
+                .withNullableObjectID("1")
+                .withNullableVersion(1)
+                .build();
+
+        when(wscli.administer(argThat(new AdminGetWSInfoAnswerMatcher(1))))
+                .thenThrow(new JsonClientException("No workspace with id 1 exists!"));
+
+        final WorkspaceEventHandler weh = new WorkspaceEventHandler(clonecli);
+
+        // update event
+        StatusEvent updatedEvent = weh.updateObjectEvent(event);
+
+        // since workspace does not exist, we expect to get back an updated event
+        Assert.assertNotSame("expected different event object", event, updatedEvent);
+        Assert.assertEquals(updatedEvent.getEventType(), StatusEventType.DELETE_ALL_VERSIONS);
+    }
+
+    @Test
+    public void updateEventObjectDeleted() throws Exception {
+        final CloneableWorkspaceClient clonecli = mock(CloneableWorkspaceClient.class);
+        final WorkspaceClient cloned = mock(WorkspaceClient.class);
+        final WorkspaceClient wscli = mock(WorkspaceClient.class);
+        when(clonecli.getClientClone()).thenReturn(cloned);
+        when(clonecli.getClient()).thenReturn(wscli);
+
+        StatusEvent event = StatusEvent.getBuilder(
+                StorageObjectType.fromNullableVersion("WS", "foo", 1),
+                Instant.ofEpochMilli(20000),
+                StatusEventType.RENAME_ALL_VERSIONS)
+                .withNullableNewName("newName")
+                .withNullableAccessGroupID(1)
+                .withNullableObjectID("1")
+                .withNullableVersion(1)
+                .build();
+
+        List<Tuple11<Long, String, String, String,
+                Long, String, Long, String, String, Long, Map<String, String>>> objList = new ArrayList<Tuple11<Long, String, String, String, Long, String, Long, String, String, Long, Map<String,String>>>();
+
+        when(wscli.administer(any()))
+                // for deleted workspace check
+                .thenReturn(new UObject(wsTuple(1L, "wsname", "username", "date", 7, "r", "n",
+                        "unlocked", Collections.emptyMap())))
+                // for deleted objects check
+                .thenReturn(new UObject(objList));
+
+        final WorkspaceEventHandler weh = new WorkspaceEventHandler(clonecli);
+
+        // update event
+        StatusEvent updatedEvent = weh.updateObjectEvent(event);
+
+        // since deleted object check fails (objid 1L is not present), we expect to get back an updated event
+        Assert.assertNotSame("expected different event object", event, updatedEvent);
+        Assert.assertEquals(updatedEvent.getEventType(), StatusEventType.DELETE_ALL_VERSIONS);
```

Comment:
Why not just build the event you expect and assertThat("event incorrect", event1, is(event2))?

Reply:
I felt it was a lot clearer to check only what was being tested but adding the entire event helps to check other fields as well. So added expected events.